### PR TITLE
docs: natspec and documentation improvements

### PR DESCRIPTION
**Pull request by gas1cent**
# 🤖 Linear

Closes OPO-639, OPO-353

### CODE REVIEW

File: docs/src/content/modules/README.md [hunk 1]
```diff
@@ -1 +1,24 @@
...
+- [Resolution](./resolution.md)
+- [Finality](./finality.md)
+
+## Common Parts
+
+You can notice that many modules follow the same structure. This is because they all inherit from the `IModule` interface which defines the common functions and modifiers that all modules should have.
+
+- `moduleName` is required to properly show the module in Prophet UI.
+- `RequestParameters` is what parameters are required for the request to be processed with the module.
+- `decodeRequestData` decodes the ABI encoded parameters using the `RequestParameters` struct and returns them as a list. This is useful for both on-chain and off-chain parts.
```
Comment (0xmoebius) [inline docs/src/content/modules/README.md:16]:
```suggestion
- `decodeRequestData` decodes the ABI encoded parameters using the `RequestParameters` struct and returns them as a list. This is useful for both on-chain and off-chain components.
```

File: docs/src/content/modules/response/bonded_response_module.md [hunk 2]
```diff
@@ -25,3 +26,4 @@ The Bonded Response Module is a contract that allows users to propose a response
 
 - **Early finalization**: It is possible for pre-dispute modules to atomically calculate the correct response on-chain, decide on the result of a dispute and finalize the request before its deadline.
 - **Dispute window**: Prevents proposers from submitting a response 1 block before the deadline and finalizing it in the next block, leaving disputers no time to dispute the response.
+- **Unutilized response**: A correct response that has not been used to finalize the request. Consider what happens when the first response to a request is disputed maliciously and someone send a second response with the same content. In that case if the second response isn't disputed and the first one comes back from the dispute and is accepted as the final response, the second proposer should be able to get their bond back.
```
Comment (0xmoebius) [inline docs/src/content/modules/response/bonded_response_module.md:29]:
```suggestion
- **Unutilized response**: A correct response that has not been used to finalize the request. Consider what happens when the first response to a request is disputed maliciously and someone sends a second response with the same content. In that case if the second response isn't disputed and the first one comes back from the dispute and is accepted as the final response, the second proposer should be able to get his bond back.
```

File: solidity/interfaces/modules/dispute/IBondEscalationModule.sol [hunk 3]
```diff
@@ -204,7 +209,9 @@ interface IBondEscalationModule is IDisputeModule {
    * @dev If this is the first dispute of the request and the bond escalation window is not over,
    *      it will start the bond escalation process. This function must be called through the Oracle.
    *
-   * @param _response  The response being disputed.
+   * @param _request The request data.
+   * @param _response The response being disputed.
+   * @param _dispute The response being disputed.
```
Comment (0xmoebius) [inline solidity/interfaces/modules/dispute/IBondEscalationModule.sol:214]: don't know how to phrase this one

File: solidity/interfaces/modules/resolution/IArbitratorModule.sol [hunk 4]
```diff
@@ -35,11 +35,14 @@ interface IArbitratorModule is IResolutionModule {
 
   /**
    * @notice Available status of the arbitration process
+   * @param Unknown The arbitration process has not started (default)
+   * @param Active The arbitration process is active
+   * @param ResolvedThe The arbitration process is finished
```
Comment (0xmoebius) [inline solidity/interfaces/modules/resolution/IArbitratorModule.sol:40]:
```suggestion
   * @param Resolved The The arbitration process is finished
```

File: solidity/interfaces/modules/resolution/IERC20ResolutionModule.sol [hunk 5]
```diff
@@ -38,6 +38,9 @@ interface IERC20ResolutionModule is IResolutionModule {
 
   /**
    * @notice Emitted when the voter gets back their bond
+   * @param _voter The voter who has claimed their bond
+   * @param _disputeId The id of the dispute
+   * @param _amount The amount of tokens claimed
    */
   event VoteClaimed(address _voter, bytes32 _disputeId, uint256 _amount);
```
Comment (0xmoebius) [inline solidity/interfaces/modules/resolution/IERC20ResolutionModule.sol:45]: we should've added the asset address to this event :P

File: solidity/interfaces/modules/dispute/IBondEscalationModule.sol (hunk 3 above)
Comment (0xmoebius) [reply solidity/interfaces/modules/dispute/IBondEscalationModule.sol:214]: `_response` desc matches the `_dispute` one

Comment (gas1cent) [reply solidity/interfaces/modules/dispute/IBondEscalationModule.sol]: Update it to match the description of the same parameter everywhere else.

**Review by 0xmoebius — APPROVED** [review]

**Pull request merged**